### PR TITLE
feat: ✨ sd-footnotes

### PR DESCRIPTION
**Pull request by mariohamann**
## Description:

BREAKING CHANGE: The `sd-footnote` component has been deprecated and removed from the library.

This PR shows adds `sd-footnotes`. As as a replacement for `sd-footnote`, it provides the possibility to use footnotes in combination with lists, which actually would be better in terms of DX and A11y.

## Definition of Reviewable:

- [ ] Documentation is created/updated
- [ ] Migration Guide is created/updated
- [ ] E2E tests (features, a11y, bug fixes) are created/updated
- [ ] Stories (features, a11y) are created/updated
- [ ] relevant tickets are linked

### CODE REVIEW

**Review by yoezlem — APPROVED** [review]

**Comment by karlbaumhauer** [comment]
@Vahid1919 pls modify the description of this PR to be marked as Breaking change.

**Review by karlbaumhauer — CHANGES_REQUESTED** [review]

File: packages/components/src/solid-styles.css [hunk 1]
```diff
@@ -13,7 +13,7 @@
 @import './styles/list/list.css';
```
[the remark this answers is not in the document]
Comment (Vahid1919) [reply packages/components/src/solid-styles.css:13]: @karlbaumhauer We do not have a migration guide for sd-footnote as it was not a part of the UI Component Library.

File: packages/components/src/styles/footnotes/footnotes.stories.ts [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+  render: (args: any) => {
+    return generateTemplate({
+      options: { templateContent: '<div class="sd-container"><ul class="%CLASSES%">%SLOT%</ul></div?' },
+      args
+    });
+  }
+};
+
+/**
+ * Use the class with a `ol` element to display the footnotes as a numbered list.
```
Comment (karlbaumhauer) [inline packages/components/src/styles/footnotes/footnotes.stories.ts:58]: @Vahid1919 what do you think about switching the "default" here? A footnote list without numbers (ul) doesn't really match the expectations of a footnote list (IMO).
maybe even with only one lingle list item as you always should add numbers if you have more. 

@coraliefeil what do you think? is there even a use case für a footnote list without numbers? 

If so, I would like to see the case specified as best as possible.

File: packages/components/src/solid-styles.css (hunk 1 above)
[the remark this answers is not in the document]
Comment (karlbaumhauer) [reply packages/components/src/solid-styles.css:13]: ups... your right. got mixed up by the libraries... my bad... 🫣

**Review by karlbaumhauer — APPROVED** [review]

**Pull request merged**